### ISSUE TITLE
How to treat  multiple valid results

### ISSUE THREAD

**Issue by julik** [WeTransfer/format_parser#21]
One of the goals we set was to be able to recognize a file as having 2 valid formats. I wonder how this could look API wise for the customer end of the library. For example, should it be a sum type?

```
info = FormatParser.parse(io)
# What should "info" be? Should it
# be an Array of file information structs
# returned by the parsers? How do we
# make case statements on this `info`
# more convenient?

case info
when :unknown # Should we implement a null object FileInformation?
when :video # What do we do if 2 parsers manage to open the file - like .mov + .mp4?
...
```

We can also make it a Hash per type, but then again - it is likely that you will get multiple `:image` natured information structs and no `:audio` natured structs.

We can also experiment with a very smart "sum type" like umbrella FileInformation that gets returned, with something like this:

```
info = FormatParser.parse(io)
info.natures #=> [:video, :image]
info.video #=> what should this return?
```

Another question is of course performance-focused - does it make sense to allow all the parsers to have a go at the file or is it reasonable to stop at the first N matches? Currently we return right on the first match, but the parsing ambiguities is one of the things we want to explore.

**Comment by tehAnswer** [comment]
**🖖**

I'm doing this one.

**Comment by bermannoah** [comment]
Closed in #47
